app/lib/discourse_automation/triggers/topic.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+DiscourseAutomation::Triggerable::TOPIC = 'topic'
+
+DiscourseAutomation::Triggerable.add(DiscourseAutomation::Triggerable::TOPIC) do
+  on_update do |automation, metadata, previous_metadata|
+    ActiveRecord::Base.transaction do
+      previous_topic_id = previous_metadata['topic_id']
+      topic_id = metadata['topic_id']
+      if previous_topic_id && previous_topic_id != topic_id
+        previous_topic = Topic.find_by(id: previous_topic_id)
+
+        if previous_topic
+          previous_topic.custom_fields.delete('discourse_automation_id')
+          previous_topic.save!
+        end
+      end
+
+      if topic_id
+        topic = Topic.find(topic_id)
```

Comment:
I believe the `find` method will trigger an exception if the topic doesn't exist. I think you want `find_by(id: ` instead.